envi/archs/arm/disasm.py

```diff
@@ -1895,12 +1895,15 @@ def repr(self, op):
             shval = shname
         else:
             shval = ""
+        #Fix putting a comma between register and shift oper
+        #may be dirty but seems to work
+        comma = ('', ',')[(len(shval))&1]                                                                                                
```

Comment:
Ahh ok so this is broke, fix this and the other line isn't needed.

        if self.shval != 0:
            shval = "%s #%d"%(shname,self.shval)
        elif self.shtype == S_RRX:
            shval = shname
        else:
            shval = ""


Date: Wed, 8 Jun 2016 11:44:17 -0700
From: notifications@github.com
To: vivisect@noreply.github.com
CC: redheadedrod@hotmail.com; author@noreply.github.com
Subject: Re: [atlas0fd00m/vivisect] Unittest (#5)

In envi/archs/arm/disasm.py:

> @@ -1895,12 +1895,15 @@ def repr(self, op):
>              shval = shname
>          else:
>              shval = ""
> +        #Fix putting a comma between register and shift oper
> +        #may be dirty but seems to work
> +        comma = ('', ',')[(len(shval))&1]                                                                                                

please move this up a few lines.  see comment there.


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.